hiero-observability/hiero-metrics-core/docs/metrics_details.md

```diff
@@ -0,0 +1,89 @@
+## Metrics Details
+
+[📘Back to Overview](metrics_overview.md)
+
+### Metric API
+
+All metrics interfaces available to use are defined in `org.hiero.metrics.api` package.<br/>
+Each interface has a builder class to configure and create an instance of the metric with specific implementation,
+that is hidden from the client. Any builder requires a [MetricKey](../src/main/java/org/hiero/metrics/api/core/MetricKey.java) to be provided,
+that identifies metric with name and class of metric interface (see [Metric Registry](#metric-registry) for more details).
+
+Main metric interface is [Metric](../src/main/java/org/hiero/metrics/api/core/Metric.java)
+which provides methods to access metric metadata and information about labels of the metric, defined during metric creation.
+Metric may have zero or more _constant_ labels that already have a value and will be attached to all data points of the metric.
+Metric may also have zero or more _dynamic_ labels, names of which are defined during metric creation
+and values are defined when accessing a specific data point of the metric for observation/update.
+Metric names and labels must be of pattern `^[a-zA-Z_][a-zA-Z0-9_]*$`.
+
+There are two main extensions of the Metric interface:
+- [StatelessMetric](../src/main/java/org/hiero/metrics/api/StatelessMetric.java)<br/>
+Terminal metric that does not provide methods to observe/update data points - only access/read them.
+Updates for such metrics are done via external code. Examples are JVM memory usage metric, CPU usage metric, etc.
+- [StatefulMetric](../src/main/java/org/hiero/metrics/api/core/StatefulMetric.java)<br/>
+This is not a terminal metric (has an abstract builder) which stores data points per unique combination of dynamic label values.
+It knows its data point type and how to instantiate it, so if new dynamic label values are provided,
+it will create a new data point and return it to the client for observation, otherwise return existing data point.
+Stateful metric also knows data point default initializer - an object used to initialize newly created data points (like initial value),
```

Comment:
Can these two processes be combined into one? Data point instantiation and initialization